patroni/postgresql/config.py

```diff
@@ -1102,8 +1102,15 @@ def _get_tcp_local_address(self) -> str:
         listen_addresses = self._server_parameters['listen_addresses'].split(',')
 
         for la in listen_addresses:
-            if la.strip().lower() in ('*', '0.0.0.0', '127.0.0.1', 'localhost'):  # we are listening on '*' or localhost
-                return 'localhost'  # connection via localhost is preferred
+            if os.name == 'nt':
+                if la.strip().lower() in ('*', 'localhost'):  # we are listening on '*' or localhost
+                    return 'localhost'  # connection via localhost is preferred
+                if la.strip().lower() in ('0.0.0.0', '127.0.0.1'):  # Postgres listens only on IPv4
+                    return '127.0.0.1'  # localhost, but don't allow Windows to resolve to IPv6
+            else:
+                # we are listening on '*' or localhost
+                if la.strip().lower() in ('*', '0.0.0.0', '127.0.0.1', 'localhost'):
+                    return 'localhost'  # connection via localhost is preferred
```

Comment:
Lets rewrite it as following to avoid code duplication:
```suggestion
            if la.strip().lower() in ('*', 'localhost'):  # we are listening on '*' or localhost
                return 'localhost'  # connection via localhost is preferred
            if la.strip().lower() in ('0.0.0.0', '127.0.0.1'):  # Postgres listens only on IPv4
                # localhost, but don't allow Windows to resolve to IPv6
                return '127.0.0.1' if os.name == 'nt' else 'localhost'
```